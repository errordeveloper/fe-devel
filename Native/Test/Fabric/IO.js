FABRIC = require('Fabric').createClient();

node = FABRIC.DependencyGraph.createNode("node");
node.addMember("resource", "FabricResource");
node.addMember("resource2", "FabricResource");

var path = FABRIC.IO.queryUserFileAndFolderHandle(FABRIC.IO.forOpenWithWriteAccess, "dummy title", "txt", "testfile");
console.log(path.fileName);

var path2 = FABRIC.IO.queryUserFileAndFolderHandle(FABRIC.IO.forOpen, "dummy title", "txt", "testfile");

try
{
  FABRIC.IO.putTextFile("dummy", "pathShouldNotBeString");
}
catch(e)
{
  console.log(e);
}

try {
  FABRIC.IO.putTextFile("dummy", path2);
}
catch (e) {
  console.log("ErrorCantWriteIfOpenMode");//Can't console.log 'e' since it contains random bytes
}

var path3 = {
  folderHandle: "badHandle",
  fileName: "dummy.txt"
  };

try {
  var badPath = FABRIC.IO.getTextFile(path3);
}
catch (e) {
  console.log(e);
}

var path4 = {
  folderHandle: path.folderHandle,
  subFolders: ["SubDir"],
  fileName: "testfile2.txt"
  };

try {
  var badPath = FABRIC.IO.getTextFile(path4);
}
catch (e) {
  console.log("DirNotFound");//Can't console.log 'e' since it contains random bytes
}

FABRIC.IO.putTextFile("FileBody", path4);
var body = FABRIC.IO.getTextFile(path4);
console.log(body);

FABRIC.IO.putUserTextFile("FileBody2", "dummyTitle", "txt", "default");
var body2 = FABRIC.IO.getUserTextFile("dummyTitle.txt", "*"); //Extension::IOManager hardcoded to get file "default.txt"
console.log(body2);

node.getResourceFromUserFile("resource", "", "txt"); //Extension::IOManager hardcoded to get file "default.txt"
console.log(JSON.stringify(node.getData("resource")));
node.putResourceToUserFile("resource", "", "txt", "testfile3");
var path5 = {
  folderHandle: path.folderHandle,
  fileName: "testfile3.txt"
  };

var body3 = FABRIC.IO.getTextFile(path5);
console.log(body3);

var path6 = {
  folderHandle: path.folderHandle,
  subFolders: ["SubDir"],
  fileName: "testfile4.txt"
  };

FABRIC.IO.putTextFile("FileBody3", path6);
node.getResourceFromFile("resource", path6);
console.log(JSON.stringify(node.getData("resource")));

var path7 = {
  folderHandle: path.folderHandle,
  subFolders: ["SubDir"],
  fileName: "testfile5.txt"
  };
node.putResourceToFile("resource", path7);
var body4 = FABRIC.IO.getTextFile(path7);
console.log(body4);

FABRIC.close();
