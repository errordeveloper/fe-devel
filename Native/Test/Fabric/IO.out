testfile.txt
Error: handleBasePath must have a 'folderHandle' member containing the handle returned by queryUserFileAndFolderHandle
ErrorCantWriteIfOpenMode
Fabric core exception: IO.getTextFile('{"path":["badHandle","dummy.txt"...'): command 'getTextFile': Unknown directory handle: 'badHandle'. Valid directory handles can only be obtained through 'IO.queryUserFileAndFolder'.
DirNotFound
FileBody
FileBody2
{"data":[70,105,108,101,66,111,100,121,50],"mimeType":"","extension":"txt","url":"default.txt"}
FileBody2
{"data":[70,105,108,101,66,111,100,121,51],"mimeType":"","extension":"txt","url":"testfile4.txt"}
FileBody3
